### PR TITLE
[GEOS-8929] Alias old com.vividsolutions.jts.geom.* to new org.locationtech.jts.geom.* in XStream reader

### PR DESCRIPTION
**Pull request by tbarsballe**
For https://osgeo-org.atlassian.net/browse/GEOS-8929

This change only affects reading; writing output is unchanged.
It is possible to alter this so that writing is likewise aliased in order to address [GEOS-8886](https://osgeo-org.atlassian.net/browse/GEOS-8886); see discussion in that ticket.

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/config/util/OneWayPackageAliasingMapper.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.geoserver.config.util;
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/config/util/OneWayPackageAliasingMapper.java]: Header?

File: src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java [hunk 2]
```diff
@@ -8,30 +8,25 @@
...
 import com.thoughtworks.xstream.io.HierarchicalStreamDriver;
 import com.thoughtworks.xstream.io.HierarchicalStreamReader;
 import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
 import com.thoughtworks.xstream.io.json.JettisonStaxWriter;
 import com.thoughtworks.xstream.io.xml.StaxWriter;
-import com.thoughtworks.xstream.mapper.CannotResolveClassException;
-import com.thoughtworks.xstream.mapper.ClassAliasingMapper;
-import com.thoughtworks.xstream.mapper.DynamicProxyMapper;
-import com.thoughtworks.xstream.mapper.Mapper;
+import com.thoughtworks.xstream.mapper.*;
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java]: Was this intensional?

Comment (tbarsballe) [reply src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java]: Nope, my IDE is being "helpful"

**Comment by tbarsballe** [comment]
Fixes pushed.

File: src/main/src/test/resources/org/geoserver/config/old_jts_binding.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            <minOccurs>0</minOccurs>
+            <maxOccurs>1</maxOccurs>
+            <nillable>false</nillable>
+            <binding>com.vividsolutions.jts.geom.LineString</binding>
+        </attribute>
+        <attribute>
+            <name>cat</name>
+            <minOccurs>0</minOccurs>
+            <maxOccurs>1</maxOccurs>
+            <nillable>false</nillable>
```
Comment (jodygarnett) [inline src/main/src/test/resources/org/geoserver/config/old_jts_binding.xml]: I would not mind seeing other bindings here in order to have a better test of bindings

Comment (jodygarnett) [reply src/main/src/test/resources/org/geoserver/config/old_jts_binding.xml]: However what you have does cover the functionality under test

**Comment by aaime** [comment]
Looks good to me

**Comment by aaime** [comment]
Ah, Travis is not happy with reason:

``
Failed to execute goal com.coveo:fmt-maven-plugin:2.4.0:check (default) on project gs-main: Found 1 non-complying files, failing build -> [Help 1]
``

**Comment by tbarsballe** [comment]
Hmm, I ran the formatter the first time around, but not after pushing my fixes; I've rerun and pushed up a version is should be happy with.

**Comment by tbarsballe** [comment]
Passing now, I've merged and backported to 2.14.x

**Pull request merged**